platforms/software/platform-base/src/main/java/org/gradle/api/plugins/BasePlugin.java

```diff
@@ -72,8 +76,16 @@ private void configureConfigurations(final Project project) {
         RoleBasedConfigurationContainerInternal configurations = (RoleBasedConfigurationContainerInternal) project.getConfigurations();
         ((ProjectInternal) project).getInternalStatus().convention("integration");
 
-        final Configuration archivesConfiguration = configurations.consumableLocked(Dependency.ARCHIVES_CONFIGURATION, conf -> {
+        final Configuration archivesConfiguration = configurations.migratingLocked(ARCHIVES_CONFIGURATION, CONSUMABLE_TO_RETIRED, conf -> {
             conf.setDescription("Configuration for archive artifacts.");
+            conf.getArtifacts().whenObjectAdded(artifact -> {
```

Comment:
This will eagerly realize all artifacts in the container.

[beforeCollectionChanges](https://github.com/gradle/gradle/blob/c6ff0875282197dadca162772fe0389923dc8d3a/subprojects/core/src/main/java/org/gradle/api/internal/DomainObjectCollectionInternal.java#L36) will do what you want and won't realize lazy elements

Reply:
Good point.  Done.